src/components/Farmhand/helpers/getPlayerInventoryQuantities.js

```diff
@@ -0,0 +1,16 @@
+import { itemsMap } from '../../../data/maps'
+
+const itemIds = Object.keys(itemsMap)
+
+/**
+ * @param {Array.<{ id: farmhand.item, quantity: number }>} inventory
+ * @returns {Object.<string, number>}
+ */
+export const getPlayerInventoryQuantities = inventory => {
+  return itemIds.reduce((acc, itemId) => {
+    const itemInInventory = inventory.find(({ id }) => id === itemId)
+    acc[itemId] = itemInInventory ? itemInInventory.quantity : 0
+
+    return acc
+  }, {})
```

Comment:
i think this would actually be more efficient as two separate operations. first one being a loop over `itemIds` to set `acc[itemId] = 0` and then a 2nd separate loop over `inventory` setting to the quantity. the way it is now, each find will loop over some portion of `inventory` that is going to be redundant.

Reply:
Good catch! This is optimized in https://github.com/jeremyckahn/farmhand/commit/3ae18d7efd8ad51e11d10dc266c30b5d9d14b4d5. It's faster AND easier to follow! :clap: